### PR TITLE
Nav placeholders, color map, styles, twig files

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# READY FOR REVIEW

# Summary
- Separating the styles related changes in the DEC-71 PR so they could be merged into master earlier.
- Also pulled in new main-nav.twig and menu-loop.twig files from #443 

# Needed By (Date)
- Soon

# Urgency
- N/A

# Steps to Test

1. Pull this branch and compile styleguide
1. Check that the Main Nav component and variants have same styles/colors as the one on the live site.
1. (for @JBCSU ). Have Redwood use this branch of Decanter, and see if the main nav still works as is with the new twig file and styles.

# Affected Projects or Products
- Decanter
- Redwood WordPress theme

# Associated Issues and/or People
- #443

### CODE REVIEW

**Comment by yvonnetangsu** [comment]
@sherakama I've added the two twig files and made minor tweaks in the scss files (mimic some of the changes you've made). I didn't just copy over your newest scss files entirely because it looks like you've been doing a lot of work on them so they've changed significantly. No need to look at this today, but if you could please take a quick look next week and let me know this is good to merge into master, that'd be awesome. Have a great weekend and thanks in advance 😄

**Comment by yvonnetangsu** [comment]
@sherakama I'll also ask JB on Monday if it makes sense to pull the secondary-nav.twig file into this PR as well.

File: core/src/scss/utilities/variables/components/_nav.scss [hunk 1]
```diff
@@ -0,0 +1,168 @@
+@charset "UTF-8";
+
+//
+// Nav Color Maps
+//
+// Color maps for the Main Nav and Secondary Nav Components.
+// "h-" prefix in the color names indicate horizontal orientation, which currently only applies to the 1st level items
+// of the desktop version of Main Nav component. Colors without the "-h" applies to the Secondary Nav (all items),
```
Comment (JBCSU) [inline core/src/scss/utilities/variables/components/_nav.scss]:
```suggestion
// of the desktop version of Main Nav component. Colors without the "h-" applies to the Secondary Nav (all items),
```

File: core/src/scss/utilities/variables/components/_nav.scss [hunk 2]
```diff
@@ -0,0 +1,168 @@
+@charset "UTF-8";
+
+//
+// Nav Color Maps
+//
+// Color maps for the Main Nav and Secondary Nav Components.
+// "h-" prefix in the color names indicate horizontal orientation, which currently only applies to the 1st level items
```
Comment (JBCSU) [inline core/src/scss/utilities/variables/components/_nav.scss]:
```suggestion
// "h-" prefix in the color names indicate horizontal orientation, which currently only applies to the top level items
```

File: core/src/scss/utilities/variables/components/_nav.scss [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+//  h-link--expanded-hover:               $color-white,
+//  h-border--expanded:                   $color-white,
+//  h-border--expanded-hover:             $color-vivid-red,
+//  h-border--expanded-active:            $color-white,
+//) !default;
+//
+// Styleguide: Variables.Components.nav-colors.variant-dark
+//
+
+$nav-dark-colors: (
```
Comment (JBCSU) [inline core/src/scss/utilities/variables/components/_nav.scss]: I would prefer a name more reminiscent of BEM naming:
```suggestion
$nav-colors--dark: (
```

File: core/src/scss/utilities/variables/components/_nav.scss [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+//  h-link--active-hover:                 $color-bright-red,
+//  h-border--active-hover:               $color-bright-red,
+//  h-link--expanded:                     $color-black,
+//  h-border--expanded:                   $color-driftwood,
+//) !default;
+//
+// Styleguide: Variables.Components.nav-colors.variant-light
+//
+
+$nav-light-colors: (
```
Comment (JBCSU) [inline core/src/scss/utilities/variables/components/_nav.scss]: I would prefer a name more reminiscent of BEM naming:
```suggestion
$nav-colors--light: (
```

File: core/src/templates/components/main-nav/main-nav.twig [hunk 5]
```diff
@@ -18,39 +18,39 @@
   <button class="su-main-nav__toggle {{ toggle_modifier_class }}" aria-expanded="false">{{ toggle_text|default('Menu') }}</button>
   <ul class="su-main-nav__menu-lv1">
     {% spaceless %}
```
Comment (JBCSU) [inline core/src/templates/components/main-nav/main-nav.twig]: I would rather not use `{% spaceless %}` around all the list items. It makes the generated markup very hard to read.

Comment (yvonnetangsu) [reply core/src/templates/components/main-nav/main-nav.twig]: I removed the spaceless tags and added a little bit of - sign trimming back in. Let me know if this is more readable.

Comment (JBCSU) [reply core/src/templates/components/main-nav/main-nav.twig]: Yes, it is more readable. Thanks!

**Comment by yvonnetangsu** [comment]
> There are a number of new Code Climate issues. Please fix as many of those as you can. Otherwise, this is looking good in Redwood!

@JBCSU I've fixed all the cc issues except the line length ones and the 2 nested selector ones.

**Comment by yvonnetangsu** [comment]
> I am fine with this.
> Understand that there will be changes in the v6 release that we hope to have this winter when #443 gets merged.

Thanks, @sherakama. And I'll start looking at your DEC-71.

**Pull request merged**